python/enthic/app.py

```diff
@@ -257,21 +263,23 @@ def page_search():
     Return a JSON formatted as page. Try to implement the Hydra
     hypermedia-driven Web APIs https://www.markus-lanthaler.com/hydra/.
     """
-
     page = int(request.args.get('page', '1')) - 1  #  TO COUNT
     if page < 0:
         ErrorJSONResponse('page parameter should be > 0')
     per_page = int(request.args.get('per_page', '30'))
     if per_page < 1:
         ErrorJSONResponse('per_page parameter should be > 0')
     probe = request.args.get('probe', "")
+    apecode = request.args.get('ape', "")
     with concurrent.futures.ThreadPoolExecutor() as executor:
-        future_list = executor.submit(result_array, probe, per_page, offset=page * per_page)
+        future_list = executor.submit(result_array, probe, per_page, ape, offset=page * per_page)
         count, = fetchone("""SELECT COUNT(*)
-                            FROM identity WHERE siren = %s
-                            OR denomination LIKE %s
-                            OR MATCH(denomination) AGAINST (%s IN NATURAL LANGUAGE MODE)""",
-                          (pre_cast_integer(probe), "{0}%".format(probe), "{0}%".format(probe)))
+                        FROM identity WHERE (siren = %s and ape regexp %s)
+                        OR (denomination LIKE %s and ape regexp %s)
+                        OR (MATCH(denomination) AGAINST (%s IN NATURAL LANGUAGE MODE) 
+                        and ape regexp %s)""", (pre_cast_integer(probe), "^{0}".format(apecode),
+                        "{0}%".format(probe), "^{0}".format(apecode), "{0}%".format(probe),
+                        "^{0}".format(ape)))
```

Comment:
You have apecode et ape and ape is not defined, probably from the copy/paste of the above change. Please respect the recommended python snake_case and keep only ape_code.

Reply:
Okay